snuba/query/allocation_policies/per_referrer.py

```diff
@@ -93,10 +93,23 @@ def _get_quota_allowance(
     ) -> QuotaAllowance:
         referrer = str(tenant_ids.get("referrer", "no_referrer"))
         concurrent_limit = self._get_concurrent_limit(referrer)
-        can_run, explanation = self._is_within_rate_limit(
+        rate_limit_params = RateLimitParameters(
+            self.rate_limit_name, referrer, None, concurrent_limit
+        )
+        rate_limit_stats = self._get_rate_limit_stats(
             query_id,
-            RateLimitParameters(self.rate_limit_name, referrer, None, concurrent_limit),
+            rate_limit_params,
         )
+        can_run, explanation = self._is_within_rate_limit(
+            rate_limit_stats,
+            rate_limit_params,
+        )
+        if not can_run:
+            self.metrics.increment(
+                "concurrent_query_exceeded_limit",
+                rate_limit_stats.concurrent,
+                tags={"referrer": referrer},
+            )
```

Comment:
This metric:

1. Should be a histogram (which in our case is a timing metric)
2. should be emitted regardless of whether the query can run
3. should be called something like "concurrent_queries_referrer"

We want to use this to basically say "how many concurrent queries is each referrer executing right now"  and max that number for every time slice.